### ISSUE TITLE
Syntax highlighting in preview mode

### ISSUE THREAD

**Issue by danisvaliev001** [antonmedv/walk#154]
It will be super if preview mode will support syntax highlighting. As I see it can be implemented using [LSP](https://microsoft.github.io/language-server-protocol/).

**Comment by antonmedv** [comment]
LSP for highlighting seems like a completely an overkill. Starting a LSP per language just for highlighting?

**Comment by danisvaliev001** [comment]
> LSP for highlighting seems like a completely an overkill. Starting a LSP per language just for highlighting?

yeah, you're right. What do you think about using [chroma library](https://github.com/alecthomas/chroma) or wrapping [bat tool](https://github.com/sharkdp/bat)?

**Comment by antonmedv** [comment]
About using chroma

**Comment by 0xTadash1** [comment]
How about using environment variables, like `WALK_REMOVE_CMD`?

I use `bat` with a custom highlighting theme and indentation size.
This approach is convenient as it removes the need to configure similar settings in `walk`.

```sh
export WALK_PREVIEW_CMD='bat'
```

**Comment by antonmedv** [comment]
Problem is with tge way bubbletea works. We have to join together two strings. Getting output and joining with walk list  is difficult.

**Comment by 0xTadash1** [comment]
lipgloss seems to support layouting.

- [Joining Paragraphs](https://github.com/charmbracelet/lipgloss?tab=readme-ov-file#joining-paragraphs)
- [Rendering Tables](https://github.com/charmbracelet/lipgloss?tab=readme-ov-file#rendering-tables)

However, using them may require not a few changes to the current implementation...

**Comment by antonmedv** [comment]
Joining i what I use now. The problem is with "spilling". Here explanation with html:
List of files:
```
file.txt
another.txt
more.txt
```

And some highlighted code:
```
yaml: <green>string
with goes multiline</green>
```

We join the together:
```
file.txt     yaml: <green>string
another.txt  with goes multiline</green>
more.txt    
```

And now `another.txt` also will be colored green.

This also happens with we truncate part of the string with "closing" escape code:
```
file.txt     yaml: <green>string | </green>
another.txt                      |
more.txt                         |
```
IF we cut stuff of screen at `|` we will color another.txt and more.txt as green too.

**Comment by danisvaliev001** [comment]
What do you think about [gocui](https://github.com/jroimartin/gocui)?
